managers/tests/test_views.py

```diff
@@ -263,12 +263,54 @@ def setUp(self):
         super().setUp()
         self.user_admin = AdminUserFactory()
         self.client.force_login(self.user_admin)
-        self.active_task_activity_1 = TaskActivityTypeFactory(is_active=True)
-        self.active_task_activity_2 = TaskActivityTypeFactory(is_active=True)
-        self.not_active_task_activity = TaskActivityTypeFactory(is_active=False)
-
         self.project = ProjectFactory()
-        self.url = reverse("project-task-activities", kwargs={"pk": self.project.pk})
+        self.task_activities = TaskActivityTypeFactory()
+        self.url = reverse("project-task-activities")
+
+    def test_manage_task_activities_view_should_get_200_when_user_is_manager_or_admin(self):
+        manager = ManagerUserFactory()
+        admin = AdminUserFactory()
+        self.client.force_login(manager)
+
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+
+        self.client.force_login(admin)
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+
+
+    def test_manage_task_activities_should_redirect_on_employee_get(self):
+        user_employee = UserFactory()
+
+        self.client.force_login(user_employee)
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, 302)
+
+    def test_manage_task_activities_view_should_add_new_task_activity_on_post_and_it_should_be_active(self):
+        task_activity_name = "New activity"
+        response = self.client.post(self.url, data={"name": task_activity_name})
+
+        self.assertEqual(response.status_code, 302)
+        self.assertIn(TaskActivityType.objects.get(name=task_activity_name), TaskActivityType.objects.filter(is_active=True))
+
+    def test_manage_task_activities_view_should_not_allow_to_create_two_tasks_with_the_same_name(self):
+
+        task_activity_name = "New activity"
+        _number_of_task_activities_at_the_start = TaskActivityType.objects.all().count()
+
+        self.client.post(self.url, data={"name": task_activity_name})
+        self.client.post(self.url, data={"name": task_activity_name})
+        _number_of_added_task_activities_if_two_taks_with_the_same_name_can_be_created = 2
+        _number_of_added_task_activities_if_only_one_task_with_the_given_name_can_be_created = 1
```

Comment:
It is only case for creating Task Activity when already existed has `is_active=True.` There is an issue that I mentioned earlier, that if Task Activity is not active (`is_active=False`), new object with the same name will be created. This shouldn't be possible. Please fix that and add some extra test case for creating new activities separately when `is_active=True` and `is_active=False`.

Reply:
Added new tests